src/osd/PG.cc

```diff
@@ -3808,14 +3808,8 @@ void PG::replica_scrub(
     return;
   }
 
-  // compensate for hobject_t's with wrong pool from sloppy hammer OSDs
-  hobject_t start = msg->start;
-  hobject_t end = msg->end;
-  start.pool = info.pgid.pool();
-  end.pool = info.pgid.pool();
-
   build_scrub_map_chunk(
-    map, start, end, msg->deep, msg->seed,
+    map, msg->start, msg->end, msg->deep, msg->seed,
```

Comment:
I'm not sure why we can remove this.  The problem was with a primary OSD running hammer and a replica running new code.  In that case, we need to specify bounds that fall within the PG (hammer would fill in some fields but not pool, which would either assert or skip items, i forget which).